x-pack/plugins/actions/server/builtin_action_types/email.ts

```diff
@@ -101,6 +103,14 @@ const ParamsSchema = schema.object(
     bcc: schema.arrayOf(schema.string(), { defaultValue: [] }),
     subject: schema.string(),
     message: schema.string(),
+    kibanaFooterLink: schema.object({
```

Comment:
this is _slightly_ weird - action params that aren't really intended to be updated by the user.

I'm guessing this was the easiest way to get the value passed in, without making more widespread changes, like adding some additional "bag" of variables/strings all over the place.

Seems fine for now, maybe we'll figure out a cleaner way to do this later.  I think it's probably worth a comment above `kibanaFooterLink` though, indicating we don't intend customers to be setting this value explicitly.

Reply:
@pmuellr ++ I'll add a comment. I agree these values won't be set by end users. Though I can still see them as parameters set programatically elsewhere like "share to" where it's done browser side and want to put a link to the visualization.

I was thinking of waiting and seeing further patterns to know what is better than params for this. Right now it's almost like hidden-from-user-while-still-being-params.

Reply:
Ya, I suspect there will be other patterns to deal with, and agree we can live with the design used in this PR till we need more.  Some things already considered:

- sending the results of a previous action into a subsequent action
- allowing customer-specified messaging "prefixes" (legal statements, etc - this kind of thing always seems to come up)

At one point I was thinking that "message" parameters should maybe be considered a list of messages, independently settable, that we'd serialize into a single message.  Something like [Slack's BlockKit blocks](https://api.slack.com/messaging/composing/layouts).